### ISSUE TITLE
Training for continuous values not working.

### ISSUE THREAD

**Issue by pepi99** [yangheng95/PyABSA#174]
Hello,

I implemented the learning for continuous targets (they are between -3.5 and 3.5).

In fast_lsa_t.py I did self.dense = nn.Linear(opt.embed_dim, 1) (because we need 1 continuous output), I removed the softmax in the apc_trainer.py and added MSELoss as loss function. But when I print the predictions and the targets, it seems that the algorithm is not learning.

Is this normal? I have 3000 sentences, is it possible that the algorithm does not have enough data to learn continuous value, or I am probably doing something wrong?

Thank you.

**Comment by yangheng95** [comment]
The reason may be that MSELoss is a unusual loss function for classification

**Comment by pepi99** [comment]
Yes, but I have transformed the problem to regression, because I am predicting continuous values.

**Comment by pepi99** [comment]
I mean, I am passing continuous values and I changed the model to output 1 continuous value (no softmax)

**Comment by yangheng95** [comment]
Maybe you can push your change to your branch, so I can try to figure out where is wrong if I got time

**Comment by pepi99** [comment]
OK. I created a new fork repo because I have problems creating the pull request: https://github.com/pepi99/pyabsa-regr

Here is a picture with the files that I changed: 
<img width="731" alt="image" src="https://user-images.githubusercontent.com/45050191/163204784-3e36caee-8b0f-40eb-9afc-082d46b7f58f.png">

**Comment by pepi99** [comment]
Basically Fast_LCF_BERT has now self.dense = nn.Linear(opt.embed_dim, 1) (only 1 output continuous value), in apc_trainer the loss is MSE (and the sentiments are not discrete, but continuous). But still, it's not learning... You can print the predictions and the targets at each epoch.

**Comment by yangheng95** [comment]
OK, I will try to make it work, but it may take a long time due to my tight schedule

**Comment by pepi99** [comment]
Ok, we can work together if you want! When are you free

**Comment by pepi99** [comment]
<img width="934" alt="image" src="https://user-images.githubusercontent.com/45050191/163344431-f818919b-872a-4639-9d4c-7156926b7a11.png">
I have changed some stuff here (float(item.polarity) and float(item[label_name]), because otherwise it is looking for discrete values and changing my polarities. Maybe it messes something with the clusters, IDK..

**Comment by yangheng95** [comment]
I am really sorry that i can not proceed on this issue due to my schedule, and this is not a widely-needed demand, so I will close this issue.
